src/components/Home.jsx

```diff
@@ -49,43 +50,92 @@ function Home() {
     }
   };
 
+  const joinListForm = () => {
```

Comment:
wow, I learned something new, actually now I see how obvious and simple this approach, I just have never used a function like this, I'd just create and  return a new component